triton-client/src/components/DatasetCard.tsx

```diff
@@ -35,10 +46,22 @@ function DatasetCard({ datasetID }: DatasetCardProps) {
 	}, [datasetID, dispatch])
 
 	const request = useCallback((downloadType: DownloadRequestType) => {
-		if (dataset) {
-			dispatch(createDownloadRequest(dataset.external_project_id, datasetID, downloadType)).catch((e) => console.error(e))
+		if (dataset && project && totalSize) {
```

Comment:
I would move this out from the callback and make the button disabled until these are true ... in this case it could allow users to click the button without result if the condition is not met when they click... If you keep it, i would make this more like a global "component is ready" state which you could reference anyplace.

Reply:
It's kinda not possible with Typescript ;)

Reply:
But I do agree with disabling the button.

Reply:
Another option is to create another subcomponent that assumes most of the states are ready.